### ISSUE TITLE
Fix shell escaping of redis/memcached password variables

### ISSUE THREAD

**Issue by timabbott** [zulip/docker-zulip#238]
A user reported seeing this output when running `docker-compose` operations for the Zulip container:

```
~/docker-zulip$ sudo docker-compose pull
[sudo] Passwort für ilyas:
WARNING: The GVqF variable is not set. Defaulting to a blank string.
Pulling database  ... done
Pulling memcached ... done
Pulling rabbitmq  ... done
Pulling redis     ... done
```

(Note that the `zulip` container isn't being run at all).

Investigating determined the problem was that their `REDIS_PASSWORD` field had some special characters in it (in this case, `$`), which resulted in Docker trying to handle it as a variable.  

`      REDIS_PASSWORD: 'E!r%ZwH3#2P$GVqF'`

See https://pastebin.com/P0jRFVnj for an example configuration with this problem.

I'm not sure what the right fix for this; at the very least we should document that passwords/secrets used in this file should use a specific character set; I'm not sure there's any escaping we can do to address this, since I think it's being considered a Docker variable.  May be worth researching what the prior art for other projects is for this issue.

**Comment by andersk** [comment]
`$VARIABLE` substitution is part of the [`docker-compose.yml` syntax](https://docs.docker.com/compose/compose-file/#variable-substitution), which also provides a syntax for escaping `$`:

> You can use a `$$` (double-dollar sign) when your configuration needs a literal dollar sign. This also prevents Compose from interpolating a value, so a `$$` allows you to refer to environment variables that you don’t want processed by Compose.
>
> ```yaml
> web:
>   build: .
>   command: "$$VAR_NOT_INTERPOLATED_BY_COMPOSE"
> ```
>
> If you forget and use a single dollar sign (`$`), Compose interprets the value as an environment variable and warns you:
>
> The VAR_NOT_INTERPOLATED_BY_COMPOSE is not set. Substituting an empty string.

**Comment by timabbott** [comment]
Yeah.  I think we should address this by updating our docs to mention not using `$` in passwords; or maybe better, we should write a little tool that substitutes in randomly generated passwords following that rule.

**Comment by i-ky** [comment]
Users can be advised to use `docker compose config` to verify that Docker Compose interprets their Compose file as intended.
